### DISCUSSION TITLE
Configuration?

### DISCUSSION THREAD

**Discussion by joegm** [artempyanykh/marksman#79] [Q&A] [answered]
Since Marksman uses a toml file as the wiki root, I suspected that might be to allow configuration somehow, but I don't see any mention or documentation of that anywhere.

This isn't a request or anything, I was just curious, but if there is no ability for per-wiki configuration then why use a toml file? Just to allow the option to support config later?

**Comment by artempyanykh** [comment, marked as answer]
That's a good question @n0s4! There are a couple reasons

1. The file is used as a project marker, similar to VCS `.git` or `.hg`, to find the *root folder*, e.g. in nvim's integration.
2. To allow configuration later, as you assumed.

I'm generally trying to avoid configuration when possible as it adds extra variability into server's behavior and makes testing harder. But #53 is a good candidate for its own configuration entry in `.marksman.toml` when I or somebody else finds the time to implement it.

**Reply by joegm** [reply]
Thanks! That clears it up.
